src/engraving/dom/score.cpp

```diff
@@ -4696,10 +4705,10 @@ ChordRest* Score::cmdNextPrevSystem(ChordRest* cr, bool next)
         // and not in first measure of entire score
         if ((destinationMeasure != firstMeasure() && destinationMeasure != firstMeasureMM())
             && (currentSegment == firstSegment || (currentMeasure->mmRest() && currentMeasure->mmRest()->isFirstInSystem()))) {
-            if (!(destinationMeasure = destinationMeasure->prevMeasure())) {
-                if (!(destinationMeasure = destinationMeasure->prevMeasureMM())) {
-                    return cr;
-                }
+            if (auto tryDestinationMeasure = destinationMeasure->prevMeasure()) {
+                destinationMeasure = tryDestinationMeasure;
+            } else if (!(destinationMeasure = destinationMeasure->prevMeasureMM())) {
+                return cr;
             }
```

Comment:
Hard to say what the original piece of code was trying to achieve here. I guess something like:
1. Try to set `destinationMeasure` using `prevMeasure`
2. If the above fails, try to set it with `prevMeasureMM` instead
3. If the above fails, return `cr`

I think there's a false assumption here which is that `prevMeasureMM` has the ability to return something non-null where `prevMeasure` can't. If you look at the implementations, for `prevMeasureMM` to succeed, there must be at least one non-null measure in the chain - and if there's one non-null measure in the chain, `prevMeasure` will have already returned it.

So i guess we can just reduce the check to:

```suggestion
        if (!(destinationMeasure = destinationMeasure->prevMeasureMM())) {
            return cr;
        }
```

Does that seem about right to you?

Reply:
I accepted this change then had second thoughts. My reading of the code is that it wants to give priority to `prevMeasure` over `prevMeasureMM`. So if what you say is true, then the code should be calling`prevMeasure` and skipping th call to `prevMeasureMM`, should it not?

Reply:
The general (simplified) flow of this method goes something like this: 

1. Set `destinationMeasure` to the first measure of the current system 
2. Set `destinationMeasure` to the previous measure (this is the bit we’re working on)
3. Set `currentSystem` to `destinationMeasure`’s system 
4. Return the first `ChordRest` in `currentSystem`  

Whether you use `prevMeasure` or `prevMeasureMM` for step 2 doesn’t really matter from a functionality standpoint - we’re only interested in `destinationMeasure`’s system which will be the same in both cases. `prevMeasure` is indeed a slightly simpler method so there is perhaps an argument that we should use it instead (I’ll leave it up to you).